src/neptune/cli/containers.py

```diff
@@ -0,0 +1,249 @@
+                if not raw_batch:
+                    break
+                version = raw_batch[-1].ver
+                batch = [element.obj for element in raw_batch]
+
+                start_time = time.monotonic()
+                expected_count = len(batch)
+                version_to_ack = version - expected_count
+                while True:
+                    try:
+                        processed_count, _ = backend.execute_operations(
+                            container_id=container_id,
+                            container_type=container_type,
+                            operations=batch,
+                            operation_storage=operation_storage,
+                        )
+                        version_to_ack += processed_count
+                        batch = batch[processed_count:]
+                        disk_queue.ack(version)
+                        if version_to_ack == version:
+                            break
+                    except NeptuneConnectionLostException as ex:
+                        if time.monotonic() - start_time > retries_timeout:
+                            raise ex
+                        logger.warning(
+                            "Experiencing connection interruptions."
+                            " Will try to reestablish communication with Neptune."
+                            " Internal exception was: %s",
+                            ex.cause.__class__.__name__,
+                        )
+
+    def move(self, *, base_path: Path, target_container_id: "UniqueId", container_type: "ContainerType") -> None:
+        new_online_dir = get_container_dir(container_id=target_container_id, container_type=container_type)
+        try:
+            (base_path / ASYNC_DIRECTORY).mkdir(parents=True, exist_ok=True)
+        except OSError:
+            logger.warning(f"Cannot create directory: {base_path / ASYNC_DIRECTORY}")
+            return
+
+        self.path.rename(base_path / ASYNC_DIRECTORY / new_online_dir)
+        self.path = base_path / ASYNC_DIRECTORY / new_online_dir
+        self.structure_version = StructureVersion.DIRECT_DIRECTORY
+
+
+@dataclass
+class Container(ABC):
+    container_id: UniqueId
+    container_type: ContainerType
+    execution_dirs: List["ExecutionDirectory"]
+    found: bool
+
+    @property
+    def synced(self) -> bool:
+        return all(map(lambda execution_dir: execution_dir.synced, self.execution_dirs))
+
+    @abstractmethod
+    def sync(self, *, base_path: Path, backend: "NeptuneBackend", project: Optional["Project"]) -> None:
+        ...
+
+    def clear(self) -> None:
+        for execution_dir in self.execution_dirs:
+            execution_dir.clear()
+
+        for execution_dir in self.execution_dirs:
+            if execution_dir.parent is not None:
+                remove_directory(execution_dir.parent)
+                break
+
+
+@dataclass
+class AsyncContainer(Container):
+    experiment: Optional["ApiExperiment"] = None
```

Comment:
Why is it optional here, since we expect it to be not `None` (line 163)? `OfflineContainer` doesn't take this parameter anyway, so we are not bound to have the same typing in those places (neither does the abstract `Container`)